File: TP3/PYTHON/Ejercicio4/Fede3p/Fede_Personaje.py
from abc import ABC, abstractmethod

class Fede_Personaje(ABC):
    
    def __init__(self, nombre, vida, nivelAtaque, nivelDefensa):
        self.__nombre = nombre
        self.__vida = vida
        self.__nivelAtaque = nivelAtaque
        self.__nivelDefensa = nivelDefensa
        
    def atacar(self):
        return self.__nivelAtaque
    
    @abstractmethod
    def defender(ataque):
        pass
    
    def getNombre(self):
        return self.__nombre
    
    def setNombre(self, nombre):
        self.__nombre = nombre
    
    def getVida(self):
        return self.__vida
    
    def setVida(self, vida):
        self.__vida = vida
        
    def getNivelDefensa(self):
        return self.__nivelDefensa
        
    def setNivelDefensa(self, nivelDefensa):
        self.__nivelDefensa = nivelDefensa
        
    def recibirDanio(self, danio):
        self.__vida = self.__vida - danio

File: TP3/JAVA/ejercicio 4/Fede/src/Fede_Personaje.java
/*Clase padre*/

public abstract class Fede_Personaje {
    
    private String nombre;
    protected Integer vida;
    protected Integer nivelAtaque;
    protected Integer nivelDefensa;

    public Fede_Personaje(String nombre, Integer vida, Integer nivelAtaque, Integer nivelDefensa){
        this.nombre = nombre;
        this.vida = vida;
        this.nivelAtaque = nivelAtaque;
        this.nivelDefensa = nivelDefensa;
    }

    public Integer atacar(){
        return nivelAtaque;
    }

    /* Esto se redefine al aplicarse */
    public abstract void defender(Integer ataque) throws VidaNegativaException;

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public Integer getVida() {
        return vida;
    }

    public void setVida(Integer vida) {
        this.vida = vida;
    }

    public void recibirDanio(Integer danio){
        this.vida = vida - danio;
    }
}
